pkg/plugins/plugins.go

```diff
@@ -122,14 +122,56 @@ func (pm *PluginManager) Init() error {
 	}
 
 	for _, p := range Plugins {
-		if !p.IsCorePlugin {
+		if p.IsCorePlugin {
+			p.Signature = PluginSignatureInternal
+		} else {
+			p.Signature = getPluginSignatureState(p.PluginDir)
 			metrics.SetPluginBuildInformation(p.Id, p.Type, p.Info.Version)
 		}
 	}
 
 	return nil
 }
 
+func getPluginSignatureState(dir string) PluginSignature {
+	manifestPath := path.Join(dir, "MANIFEST.txt")
+	info, _ := os.Stat(manifestPath)
+	if info == nil || info.Size() < 5 {
+		return PluginSignatureUnsigned
+	}
+
+	// 1. TODO: validate PGP signature
+	// if( !valid ) {
+	//   return PluginSignatureInvalid
+	// }
+
+	// 2. check the manifest contents... currently looks like:
+	//
```

Comment:
Any thoughts on a format here?  This could easily be a JSON object...

Also, I think we should remove unimportant files from the manifest (*.png) so we don't need to validate them